cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/SwitchesChain.java

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.cgmes.conversion;
+
+import java.util.*;
+
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Connectable;
+import com.powsybl.iidm.network.IdentifiableType;
+import com.powsybl.iidm.network.Switch;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.VoltageLevel;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+class SwitchesChain {
+    private final NodeConnectables nodeConnectables;
+    private final Adjacency adjacency;
+    private final VoltageLevel vl;
+    private final Switch sw;
+    private final boolean isNodeBreakerModel;
+
+    SwitchesChain(VoltageLevel vl, Switch sw, boolean isNodeBreakerModel) {
+        this.vl = vl;
+        this.sw = sw;
+        this.isNodeBreakerModel = isNodeBreakerModel;
```

Comment:
Trying to have the same code for the two kinds of view is a bit artificial, I'm not sure the benefit is real here.

Reply:
I refactored the code. I think now is more readable.